src/components/Popup/Popup.tsx

```diff
@@ -66,6 +68,36 @@ export default class Popup extends AutoControlledComponent<Extendable<PopupProps
 
   public static Content = PopupContent
 
+  // React bindings for autocontrolled component are introduced
+  private stateManager = popupStateManager.withBindings({
+    /* derives state from component */
+    getState: () => {
+      // AUTOCONTROLLED state logic
+      // note, sloppy implementation, but just to provide an idea:
+      // - 'state' that is seen by state manager could be composed from different data sources in the way necessary
+      //
+      // edge cases of null/undefined are not covered, but this is pretty easy implementation task
+      return { ...this.state, ...this.props }
+    },
+
+    /*
+    * AUTOCONTROLLED implementation for absolute state match.
+    * - acts like a filter for the state changes that should be applied
+    */
+    // -------------
+    // willSetState: ({ prevState, stateDiff, args, actionName }) => {
+    //   return excludeControlledProps(stateDiff, this.props)
+    // },
+
+    setState: ({ stateDiff, prevState, newState, actionName, userArgs }) => {
+      const { eventArgs } = userArgs
+
+      // although state doesn't precisely reflect autocontrolled 'state',
+      // the state of stateManager is always properly defined
+      this.setState(stateDiff, () => _.invoke(this.props, 'onOpenChange', eventArgs, newState))
```

Comment:
`actionName` could be tested as well, to guarantee proper event will be triggered